### PR TITLE
feat: refactoring and type checking operators

### PR DESCRIPTION
**Pull request by aljazerzen**
- fix: lineage of unnamed columns
- fix
- fix
- refactor: resolver
- refactor
- TableDecl.column RelationColumn -> TupleField
- TableDecl columns -> ty
- renaming relations
- refactor: BinOp, UnOp -> RqOperator
- typecheck operators
- types types types
- tests, fmt and clippy

### CODE REVIEW

File: prql-compiler/src/semantic/static_analysis.rs [hunk 1]
```diff
@@ -1,79 +1,82 @@
...
-            if let Some(lit) = res {
-                ExprKind::Literal(lit)
-            } else if let (BinOp::Coalesce, ExprKind::Literal(Literal::Null)) = (op, &left.kind) {
-                right.kind
-            } else {
-                ExprKind::Binary { left, op, right }
-            }
+                _ => {}
+            };
+            ExprKind::RqOperator { name, args }
```
Comment (max-sixty) [inline prql-compiler/src/semantic/static_analysis.rs:79]: Very nice file!

File: prql-compiler/src/semantic/std.prql [hunk 2]
```diff
@@ -10,9 +10,31 @@
...
+let sub = left<int || float || timestamp || date> right<int || float || timestamp || date> -> <int || float || timestamp || date> internal std.sub
+let eq = left right -> <bool> internal std.eq
+let ne = left right -> <bool> internal std.ne
+let gt = left right -> <bool> internal std.gt
+let lt = left right -> <bool> internal std.lt
+let gte = left right -> <bool> internal std.gte
+let lte = left right -> <bool> internal std.lte
+let and = left<bool> right<bool> -> <bool> internal std.regex_search
+let or = left<bool> right<bool> -> <bool> internal std.and
+let coalesce = left right -> internal std.or
```
Comment (max-sixty) [inline prql-compiler/src/semantic/std.prql:29]: Great!

File: prql-compiler/src/tests/test_error_messages.rs [hunk 3]
```diff
@@ -179,3 +179,20 @@ fn test_bad_function_type() {
...
+    from foo
+    select (a && b) + c
+    "#)
+    .unwrap_err(), @r###"
+    Error:
+       ╭─[:3:13]
+       │
+     3 │     select (a && b) + c
+       │             ───┬──
+       │                ╰──── function std.add, param `left` expected type `int || float || timestamp || date`, but found type `bool`
```
Comment (max-sixty) [inline prql-compiler/src/tests/test_error_messages.rs:195]: Very impressive...

File: web/book/src/transforms/window.md [hunk 4]
```diff
@@ -84,13 +84,13 @@ You can also only apply `group`:
...
   sort age
-  derive rnk = rank
+  derive {rnk = rank age}
 )
 ```
 
 ## Window functions as first class citizens
 
-There is no limitaions where windowed expressions can be used:
+There is no limitations where windowed expressions can be used:
```
Comment (max-sixty) [inline web/book/src/transforms/window.md]:
```suggestion
There are no limitations on where windowed expressions can be used:
```

Comment (max-sixty) [reply web/book/src/transforms/window.md]: @aljazerzen  why not accept these?

Comment (aljazerzen) [reply web/book/src/transforms/window.md]: I did accept them?

Comment (max-sixty) [reply web/book/src/transforms/window.md]: Very sorry, I must have misread, and was thinking whether they were too refine-y...

**Comment by max-sixty** [comment]
Somewhere here we're getting a stack overflow on Windows: https://github.com/PRQL/prql/actions/runs/5164153795/jobs/9304807161

**Pull request merged**